addons/website_sale/models/sale_order.py

```diff
@@ -409,3 +418,38 @@ def _get_shop_warning(self, clear=True):
         if clear:
             self.shop_warning = ''
         return warn
+
+    def _filter_can_send_abandoned_cart_mail(self, website):
+        abandoned_datetime = datetime.utcnow() - relativedelta(hours=website.cart_abandoned_delay)
+
+        sales_done_after_self_create_date = self.env['sale.order'].search([
+            ('state', '=', 'sale'),
```

Comment:
Just in case, would add the condition to exclude self in the resultset.
```suggestion
            ('id', 'not in', self.ids),
            ('state', '=', 'sale'),
```